pkg/server/server.go

```diff
@@ -313,23 +301,65 @@ func (s *Server) Run(ctx context.Context) error {
 		workspaceLister = kcpSharedInformerFactory.Tenancy().V1alpha1().Workspaces().Lister()
 	}
 
-	serverOptions.APIExtensionsNewSharedInformerFactoryFunc = func(client apiextensionsclient.Interface, resyncPeriod time.Duration) apiextensionsexternalversions.SharedInformerFactory {
+	completedOptions, err := genericcontrolplane.Complete(serverOptions)
+	if err != nil {
+		return err
+	}
+
+	apisConfig, _, pluginInitializer, err := genericcontrolplane.CreateKubeAPIServerConfig(completedOptions)
+	if err != nil {
+		return err
+	}
+
+	// Wire in a ServiceResolver that always returns an error that ResolveEndpoint is not yet
+	// supported. The effect is that CRD webhook conversions are not supported and will always get an
+	// error.
+	serviceResolver := &unimplementedServiceResolver{}
+
+	// If additional API servers are added, they should be gated.
+	apiExtensionsConfig, err := genericcontrolplane.CreateAPIExtensionsConfig(
+		*apisConfig.GenericConfig,
+		apisConfig.ExtraConfig.VersionedInformers,
+		pluginInitializer,
+		completedOptions.ServerRunOptions,
+		serviceResolver,
+		webhook.NewDefaultAuthenticationInfoResolverWrapper(
+			nil,
+			apisConfig.GenericConfig.EgressSelector,
+			apisConfig.GenericConfig.LoopbackClientConfig,
+			apisConfig.GenericConfig.TracerProvider,
+		),
+	)
+	if err != nil {
+		return fmt.Errorf("configure api extensions: %v", err)
+	}
+
+	apiExtensionsConfig.ExtraConfig.NewInformerFactoryFunc = func(client apiextensionsclient.Interface, resyncPeriod time.Duration) apiextensionsexternalversions.SharedInformerFactory {
 		f := apiextensionsexternalversions.NewSharedInformerFactory(client, resyncPeriod)
 		return &kcpAPIExtensionsSharedInformerFactory{
 			SharedInformerFactory: f,
 			workspaceLister:       workspaceLister,
 		}
 	}
```

Comment:
@ncdc this is much more in the philosophy of kube plumbing, i.e. in configs you override, not in options (which are just variables for flags (with very few exceptions)).